File: P Olimpico de Programação (G4M3) IFPB/quadradoperfeito.py
def bs(inicio, fim, valor):
  meio = (fim + inicio) // 2
  if(meio * meio == valor):
    return meio
  elif meio == inicio or meio == fim:
    return -1
  elif meio * meio > valor :
    return bs(inicio, meio, valor)
  return bs(meio, fim, valor)

n = int(input())

if bs(1, n, n) == -1:
  print("NAO")
else:
  print("SIM")

File: Programa Olimpico de Programação IFPB/quadradoperfeito.cpp
#include <bits/stdc++.h>
using namespace std;

int bs(int inicio, int fim, int valor){
  float meio = ceil((fim + inicio)/2);
  if(meio * meio == valor) return meio;
  else if ((meio == inicio) || (meio == fim)) return -1;
  else if (meio * meio > valor) return bs(inicio, meio, valor);
  return bs(meio, fim, valor);
}
int main() {
    int n; cin >> n;
    if (bs(1, n, n) == -1) cout << "NAO";
    else cout << "SIM";
    return 0;
}